helsingborg-stad/helsingborg-io-sls-api
# Fix/recurring application

Apologize in advance for the large PR :-)

Just ask if someting is hard to understand.

## Solution description
Refactored viva-ms lambdas to handle when workflow id is absent. Added two new lambdas to handle viva application status and viva application person info.
- personApplicationStatus - checks if applicant is allowed to apply and if a new period is open
- personApplication - get applicant person information to be saved onto the case item in dynamoDB

Below flow chart illustrate the most important event flow
![Event flow schema](https://t2619042.p.clickup-attachments.com/t2619042/1da067a9-896b-42bf-ad2a-4c09b9c2bef7/CaseStatus.jpg?view=open "Event Flow schema")

More info can be found on the task in ClickUp.

## What areas is affected by these changes?.
viva-ms
pdf-ms (deprecated service but used in current production environment)
html-to-pdf

## Is there any exsisting behavior change of other features due to this code change?
To have control where in the proccess an form is, a new attribute is added to the case - 'state'
Lambdas will update state during the life time of the case to make other lambdas react at the correct phase.

## How to test?
1. Check out this branch
2. Deploy viva-ms, pdf-ms and html-to-pdf to your AWS sandbox
3. Have an open period ready for one applicant (Viva)
4. Use the app and log in
5. Verify that an new case is open
6. Finish the application and submit it
7. Complete the Viva administation part ie. approve or decline the application in Viva
8. Log out
9. Log in
10. Verify that an new period is open


## Was this feature tested in the following environments?
- [x] App (iPhone)
